xdsl/dialects/linalg.py

```diff
@@ -752,6 +752,116 @@ class Conv2DNchwFchwOp(ConvOpsBase):
     name = "linalg.conv_2d_nchw_fchw"
 
 
+@irdl_op_definition
+class BroadcastOp(IRDLOperation):
+    """
+    Static broadcast operator
+
+    Broadcast the input into the given shape by adding dimensions
+    """
+
+    name = "linalg.broadcast"
+
+    input = operand_def(MemRefType | AnyTensorType)
+    init = operand_def(MemRefType | AnyTensorType)
+    result = var_result_def(AnyTensorType)
+
+    dimensions = attr_def(DenseArrayBase)
+
+    def __init__(
+        self,
+        input: SSAValue,
+        init: SSAValue,
+        dimensions: Attribute,
+        result: Attribute | None = None,
+    ):
+        super().__init__(
+            attributes={
+                "dimensions": dimensions,
+            },
+            operands=(input, init),
+            result_types=(result,),
+        )
+
+    def verify_(self) -> None:
+
+        assert isinstance(input_type := self.input.type, TensorType | MemRefType)
+        assert isinstance(init_type := self.init.type, TensorType | MemRefType)
+
+        dimensions_shape = cast(list[int], self.dimensions.as_tuple())
```

Comment:
this is spooky, is it a tuple or a list?

Reply:
ill remove that, the casting is neccessary here